coops-ui/tests/unit/AddressListSm.spec.ts

```diff
@@ -4,79 +4,132 @@ import Vuelidate from 'vuelidate'
-      expect(vm.deliveryAddress).toBeNull()
-      expect(vm.$el.querySelectorAll('.v-list-item').length).toEqual(1)
-      expect(vm.$el.querySelector('.v-list-item__title').textContent).toBe('Mailing Address')
-      expect(vm.$el.querySelector('.address-info').textContent).toContain('Victoria BC')
+    const wrapper = mount(AddressListSm, { store, vuetify })
+    const vm = wrapper.vm as any
 
+    Vue.nextTick(async () => {
+      expect(vm.registeredAddress).toBeDefined()
+      expect(vm.recordsAddress).toBeNull()
+      expect(vm.$el.querySelector(
+        '.address-panel .address-info').textContent).toContain('Glasgow')
+      expect(vm.$el.querySelector(
+        '.address-panel .address-info li:nth-child(4)').textContent).toContain('Canada')
+
+      wrapper.destroy()
       done()
     })
   })
 
-  it('displays the delivery address', done => {
-    // init store
-    store.state.mailingAddress = null
-    store.state.deliveryAddress = {
-      'streetAddress': '220 Buchanan St',
-      'addressCity': 'Glasgow',
-      'addressRegion': 'Scotland',
-      'postalCode': 'G1 2FFF',
-      'addressCountry': 'UK'
+  it('displays registered address and records address data when a Bcorp', async done => {
+    // Init Store
+    store.state.entityType = EntityTypes.BCorp
+    store.state.registeredAddress = {
+      'deliveryAddress':
+        {
+          'streetAddress': '220 Buchanan St',
+          'addressCity': 'Glasgow',
+          'addressRegion': 'Scotland',
+          'postalCode': 'G1 2FFF',
+          'addressCountry': 'CA'
+        },
+      'mailingAddress':
+        {
+          'streetAddress': '220 Buchanan St',
+          'addressCity': 'Glasgow',
+          'addressRegion': 'Scotland',
+          'postalCode': 'G1 2FFF',
+          'addressCountry': 'CA'
+        }
     }
+    store.state.recordsAddress = {
+      'deliveryAddress':
+        {
+          'streetAddress': '123 Cloverdale St',
+          'addressCity': 'Victoria',
+          'addressRegion': 'BC',
+          'postalCode': 'G1 2FFF',
+          'addressCountry': 'CA'
+        },
+      'mailingAddress':
+        {
+          'streetAddress': '321 Burrard St',
+          'addressCity': 'Vancouver',
+          'addressRegion': 'BC',
+          'postalCode': 'G1 2FFF',
+          'addressCountry': 'CA'
+        }
+    }
+
+    const wrapper = mount(AddressListSm, { store, vuetify })
+    const vm = wrapper.vm as any
+
+    // Click the records office tab to display the addresses
+    const button = vm.$el.querySelector('#record-office-panel')
+    await button.click()
+
+    Vue.nextTick(async () => {
+      expect(vm.registeredAddress).toBeDefined()
+      expect(vm.recordsAddress).toBeDefined()
+
+      expect(vm.$el.querySelector(
+        '.address-panel .address-info').textContent).toContain('Glasgow')
+      expect(vm.$el.querySelector(
+        '.address-panel .address-info li:nth-child(4)').textContent).toContain('Canada')
```

Comment:
For future: this kind of selector usage should be replaced by an attribute like
`data-test-id="record-delivery-address"`
possibly on the v-list-item or something like this.

Reply:
I definitely agree. Was sort of waiting on seeing how the unique attributes will be implemented for the front end E2E testing.

Reply:
How unique attributes should be implemented is described in the attached doc in https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/1721

But this can be done in a future commit (not this PR)... resolving for now.

Reply:
Yes i agree completely, hopefully these changes can be made all at once to create consistency opposed to a little here and a little there.